### ISSUE TITLE
Research - After ring all reduce, how can we backprop when we are manually computing the gradients across workers?

### ISSUE THREAD

**Issue by ClawCodes** [ClawCodes/DistNet#4]
This [PyTorch Page](https://docs.pytorch.org/tutorials/intermediate/ddp_tutorial.html) states 

"To make this happen, DDP registers an autograd hook for each parameter in the model. When the backward pass is run, this hook fires and triggers gradient synchronization across all processes. This ensures that each process has the same gradients, which are then used to update the model."

Can we take the same approach?

[This page](https://docs.pytorch.org/tutorials/intermediate/autograd_saved_tensors_hooks_tutorial.html) might be relevant for creating a hook.
